packages/slate-editor/src/extensions/heading/components/HeadingElement.tsx

```diff
@@ -12,15 +15,48 @@ interface Props extends HTMLAttributes<HTMLHeadingElement> {
 
 export const HeadingElement = forwardRef<HTMLHeadingElement, Props>(
     ({ children, className, element, ...props }, ref) => {
+        const editor = useSlateStatic();
         const Heading = element.type === HEADING_1_NODE_TYPE ? 'h3' : 'h4';
+
+        const isHeadingOne = element.type === HEADING_1_NODE_TYPE;
+        const isHeadingTwo = element.type === HEADING_2_NODE_TYPE;
+        const isTitle = element.role === HeadingRole.TITLE;
+        const isSubtitle = element.role === HeadingRole.SUBTITLE;
+
+        const finalClassNames = classNames(className, styles.HeadingElement, {
+            [styles.title]: isTitle,
+            [styles.subtitle]: isSubtitle,
+            [styles.headingOne]: !isTitle && isHeadingOne,
+            [styles.headingTwo]: !isSubtitle && isHeadingTwo,
+        });
+
+        if (isTitle && EditorCommands.isNodeEmpty(editor, element)) {
```

Comment:
I've considered using Slate's `renderPlaceholder` but that only seems to work when the editor is completely empty.

This way, we always render the placeholder as long as the nodes are empty.